models/glif_psc.cpp

```diff
@@ -0,0 +1,578 @@
+}
+
+/* ----------------------------------------------------------------
+ * Parameter and state extractions and manipulation functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::glif_psc::Parameters_::get( DictionaryDatum& d ) const
+{
+  def< double >( d, names::V_th, th_inf_ + E_L_ );
+  def< double >( d, names::g, G_ );
+  def< double >( d, names::E_L, E_L_ );
+  def< double >( d, names::C_m, C_m_ );
+  def< double >( d, names::t_ref, t_ref_ );
+  def< double >( d, names::V_reset, V_reset_ + E_L_ );
+
+  def< double >( d, names::a_spike, a_spike_ );
+  def< double >( d, names::b_spike, b_spike_ );
+  def< double >( d, names::a_reset, voltage_reset_a_ );
+  def< double >( d, names::b_reset, voltage_reset_b_ );
+
+  def< double >( d, names::a_voltage, a_voltage_ );
+  def< double >( d, names::b_voltage, b_voltage_ );
+
+  def< std::vector< double > >( d, names::asc_init, asc_init_ );
+  def< std::vector< double > >( d, names::k, k_ );
+  def< std::vector< double > >( d, names::asc_amps, asc_amps_ );
+  def< std::vector< double > >( d, names::r, r_ );
+  ArrayDatum tau_syn_ad( tau_syn_ );
+  def< ArrayDatum >( d, names::tau_syn, tau_syn_ad );
+  def< bool >( d, names::has_connections, has_connections_ );
+  def< model_type >( d, names::glif_model, glif_model_ );
+}
+
+double
+nest::glif_psc::Parameters_::set( const DictionaryDatum& d )
+{
+  // if E_L_ is changed, we need to adjust all variables defined relative to
+  // E_L_
+  const double ELold = E_L_;
+  updateValue< double >( d, names::E_L, E_L_ );
+  const double delta_EL = E_L_ - ELold;
+
+  if ( updateValue< double >( d, names::V_reset, V_reset_ ) )
+  {
+    V_reset_ -= E_L_;
+  }
+  else
+  {
+    V_reset_ -= delta_EL;
+  }
+
+  if ( updateValue< double >( d, names::V_th, th_inf_ ) )
+  {
+    th_inf_ -= E_L_;
+  }
+  else
+  {
+    th_inf_ -= delta_EL;
+  }
+
+  updateValue< double >( d, names::g, G_ );
+  updateValue< double >( d, names::C_m, C_m_ );
+  updateValue< double >( d, names::t_ref, t_ref_ );
+
+  updateValue< double >( d, names::a_spike, a_spike_ );
+  updateValue< double >( d, names::b_spike, b_spike_ );
+  updateValue< double >( d, names::a_reset, voltage_reset_a_ );
+  updateValue< double >( d, names::b_reset, voltage_reset_b_ );
+
+  updateValue< double >( d, names::a_voltage, a_voltage_ );
+  updateValue< double >( d, names::b_voltage, b_voltage_ );
+
+  updateValue< std::vector< double > >( d, names::asc_init, asc_init_ );
+  updateValue< std::vector< double > >( d, names::k, k_ );
+  updateValue< std::vector< double > >( d, names::asc_amps, asc_amps_ );
+  updateValue< std::vector< double > >( d, names::r, r_ );
+
+  updateValue< std::vector< double > >( d, names::tau_syn, tau_syn_ );
+  updateValue< model_type >( d, names::glif_model, glif_model_ );
+
+  // check after ASC parameters size
+  if ( not(
+         ( asc_init_.size() == k_.size() ) && ( k_.size() == asc_amps_.size() ) && ( asc_amps_.size() == r_.size() ) ) )
```

Comment:
Please use `and` and `or` instead of `&&` and `||` for boolean operators.

I also think the code would be clearer if you chose one value as reference and then compared all others to it, e.g.,

```c++
const size_t asc_size = asc_init_.size();
if ( not ( k_.size() == asc_size and asc_amps._size() == asc_size and ... ) )
```